cdap-data-fabric/src/test/java/co/cask/cdap/data2/dataset2/lib/table/MetadataStoreDatasetTest.java

```diff
@@ -84,6 +92,14 @@ public boolean apply(Integer input) {
       Assert.assertEquals(i, (int) Iterables.get(val.values(), 0));
       Assert.assertEquals(i + 1, (int) Iterables.get(val.values(), 1));
     }
+
+    // Fetch all keys using keySet
+    Map<MDSKey, Integer> val = metadataStoreDataset.listKV(keySet, Integer.class, 5);
```

Comment:
Again would be good to add another assertion to fetch fewer than 100% of the keys in the table